src/commands/protocols/get/sender/network-get-command.js

```diff
@@ -5,13 +5,28 @@ class NetworkGetCommand extends NetworkProtocolCommand {
     constructor(ctx) {
         super(ctx);
         this.operationService = ctx.getService;
+        this.ualService = ctx.ualService;
+        this.validationModuleManager = ctx.validationModuleManager;
 
         this.errorType = ERROR_TYPE.GET.GET_NETWORK_ERROR;
     }
 
-    getKeywords(command) {
-        const { assertionId } = command.data;
-        return [assertionId];
+    // TODO: discuss if we want to change to Hash(contract/tokenId/aId1) for the locationHash
+    async getKeywords(command) {
+        const { blockchain, contract, tokenId } = command.data;
+        const ual = this.ualService.deriveUAL(blockchain, contract, tokenId);
+        const firstAssertionId = await this.blockchainModuleManager.getAssertionByIndex(
+            blockchain,
+            contract,
+            tokenId,
+            0,
+        );
+        const locationHash = await this.validationModuleManager.callHashFunction(
```

Comment:
Can you move this code in method, something like calculateLocationHash and put it in ualService?